libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -400,40 +400,54 @@ void AP_UAVCAN::do_cyclic(void)
-                         * physically possible throws at [-1:1] limits.
-                         */
+                        do {
+                            repeat_send = false;
+                            uavcan::equipment::actuator::ArrayCommand msg;
 
-                        if (_rco_conf[i].active && ((((uint32_t) 1) << i) & _servo_bm)) {
-                            cmd.actuator_id = i + 1;
+                            uint8_t k = 0;
+                            for (uint8_t i = 0; i < UAVCAN_RCO_NUMBER && k < 15; i++) {
+                                uavcan::equipment::actuator::Command cmd;
 
-                            // TODO: other types
-                            cmd.command_type = uavcan::equipment::actuator::Command::COMMAND_TYPE_UNITLESS;
+                                /*
+                                 * Servo output uses a range of 1000-2000 PWM for scaling.
+                                 * This converts output PWM from [1000:2000] range to [-1:1] range that
+                                 * is passed to servo as unitless type via UAVCAN.
+                                 * This approach allows for MIN/TRIM/MAX values to be used fully on
+                                 * autopilot side and for servo it should have the setup to provide maximum
+                                 * physically possible throws at [-1:1] limits.
+                                 */
 
-                            // TODO: failsafe, safety
-                            cmd.command_value = constrain_value(((float) _rco_conf[i].pulse - 1000.0) / 500.0 - 1.0, -1.0, 1.0);
+                                if (_rco_conf[i + starting_servo].active && ((((uint32_t) 1) << (i + starting_servo)) & _servo_bm)) {
+                                    cmd.actuator_id = (i + starting_servo) + 1;
 
-                            msg.commands.push_back(cmd);
+                                    // TODO: other types
+                                    cmd.command_type = uavcan::equipment::actuator::Command::COMMAND_TYPE_UNITLESS;
 
-                            // UAVCAN can hold maximum of 15 commands in one frame
-                            k++;
-                        }
-                    }
+                                    // TODO: failsafe, safety
+                                    cmd.command_value = constrain_value(((float) _rco_conf[i + starting_servo].pulse - 1000.0) / 500.0 - 1.0, -1.0, 1.0);
+
+                                    msg.commands.push_back(cmd);
+
+                                    // UAVCAN can hold maximum of 15 commands in one frame
+                                    k++;
+                                }
+                            }
+
+                            if (msg.commands.size() > 0) {
+                                act_out_array->broadcast(msg);
 
-                    if (msg.commands.size() > 0) {
-                        act_out_array->broadcast(msg);
+                                if(k == 15) {
```

Comment:
This needs to be 14, `k` will never reach 15.

By the way, it would be nicer to remove `k` and just use `msg.commands.size()`, but I'm fine with either way.